sdks/java/extensions/sbe/src/main/java/org/apache/beam/sdk/extensions/sbe/SbeFieldUtils.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.extensions.sbe;
+
+import org.apache.beam.sdk.extensions.sbe.SbeSchema.UnsignedOptions;
+import org.apache.beam.sdk.schemas.Schema.FieldType;
+import uk.co.real_logic.sbe.PrimitiveType;
+
+/** Utilities for working with SBE fields. */
+final class SbeFieldUtils {
+  private SbeFieldUtils() {}
+
+  /**
+   * Converts a SBE {@link PrimitiveType} to a Beam {@link FieldType}.
+   *
+   * <p>"Signed" in this case simply means that it isn't one of the unsigned types (uint8, uint16,
+   * uint32, uint64). This includes the char type.
+   *
+   * @param sbePrimitive the signed {@link PrimitiveType}
+   * @return the Beam equivalent of the SBE type
+   */
+  static FieldType signedSbePrimitiveToBeamPrimitive(PrimitiveType sbePrimitive) {
```

Comment:
I'd prefer if these were just private helpers (or even inlined) in `PrimitiveSbeField`. It looks like they're only called there and in tests. The tests could just verify the higher-level public API instead.